net-istio-controller/rockcraft.yaml

```diff
@@ -0,0 +1,53 @@
+# From (ko image): https://github.com/knative-extensions/net-istio/tree/release-1.12
+name: net-istio-controller
+base: ubuntu@22.04
+version: v1.12.3
+summary: An image for Knative's net-istio-controller
+description: |
+  An image for Knative's net-istio-controller
+license: Apache-2.0
+entrypoint-service: net-istio-controller
+run-user: _daemon_
+
+platforms:
+  amd64:
+
+services:
+  net-istio-controller:
+    override: replace
+    command: "/ko-app/controller [ ]"
+    startup: enabled
+    user: ubuntu
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
+      dpkg-query --root=${CRAFT_PROJECT_DIR}/../bundles/ubuntu-22.04/rootfs/ -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
+      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  net-istio-controller:
+    plugin: go
+    build-snaps:
+      - go/1.21/stable
+    source: https://github.com/knative-extensions/net-istio.git
+    source-tag: release-1.12
```

Comment:
I prefer to use the tag `knative-v1.12.3` from the specific minor release over this branch, to keep consistency of the built rock in case a new minor version was introduced